feat: add article integration crm (#1042)
eleven-labs/blog.eleven-labs.com: _articles/fr/2023-11-29-integration-crm.md

---
contentType: article
lang: fr
date: 2023-11-29
slug: integration-crm
title: "Comment intégrer son CRM en asynchrone"

> excerpt: "Cas d'usage avec Hubspot et RabbitMQ"

Reviewer: ```suggestion
excerpt: "Découvrez un cas d'usage d'intégration d'un CRM en asynchrone, avec Hubspot et RabbitMq, au sein d'une application e-commerce"
```

categories:
    - php
keywords:
    - crm
    - hubspot
    - rabbitmq
authors:
    - youyou
    - fbelet
    - charles-eric
---
Dans cet article, nous allons nous plonger dans l'intégration d'un CRM (Customer Relationship Management)
au sein d'une application e-commerce.

> Un CRM est une solution SaaS (Software as a Service) externe dotée de fonctionnalités,

Reviewer 2: > CRM est une solution SaaS

pas nécessairement, tu peux installer un CRM sur ton propre serveur comme SugerCRM par exemple.

```suggestion
Le CRM que nous utilisons est en SaaS (Software as a Service) et est dotée de fonctionnalités,
```

telles qu'une API et des webhooks (qui permettent de recevoir des notifications en temps réel).
Ces fonctionnalités jouent un rôle central dans la gestion efficace des relations clients.

Pour illustrer ce processus, nous allons nous concentrer sur une application e-commerce personnalisée
qui représente le cœur de notre activité.

Bien entendu, ces principes d'intégration pourraient s’appliquer également à divers systèmes ERP
(Enterprise Resource Planning), mais pour les besoins de notre exemple, nous allons prendre en considération
deux CRM renommés, à savoir Salesforce et HubSpot. (Restez à l'affût, car nous publierons bientôt des
articles détaillés sur l'intégration avec chacun de ces deux CRM.)

![Schema Global]({BASE_URL}/imgs/articles/2023-11-29-integration-crm/integration-crm-schema-global.png)

> Pour réaliser cette intégration, nous allons adopter une approche basée sur le concept d'« event-driven »,

Reviewer: ```suggestion
Pour réaliser cette intégration, nous allons adopter une approche basée sur le concept d'**event-driven**,
```

où des événements (events) se déclenchent des deux côtés de l'équation,
et nous utiliserons l'outil RabbitMQ pour faciliter cette synchronisation bidirectionnelle.

Par exemple, des événements peuvent provenir de notre plateforme

> (comme 'business_created' ou 'purchase_updated') et aussi du CRM (comme 'company_edited').

Reviewer: ```suggestion
(comme `business_created` ou `purchase_updated`) et aussi du CRM (comme `company_edited`).
```

Les événements se manifestent des deux côtés.

Je vous invite à consulter cet [article de Marie](https://blog.eleven-labs.com/fr/event-driven-architecture-examples/) qui explore plus en profondeur
le concept d'« event-driven ».

> ![Schema Events]({BASE_URL}/imgs/articles/2023-11-29-integration-crm/integration-crm-schema-rabbit.png)

Reviewer 2: tu peux le faire en mermaid le schéma :D

```suggestion
```mermaid
flowchart LR
    CRM --event company_created--> E_COMMERCE["E-Commerce"] --event user_created--> CRM
```

Lorsque l'on synchronise un CRM avec une application e-commerce,
il est essentiel de prendre en compte des correspondances.

> Par exemple, un 'account' côté CRM équivaut à un 'user' côté e-commerce,
> un 'order' à un 'purchase', et un 'company' à un 'business'."

Reviewer: ```suggestion
Par exemple, un `account` côté CRM équivaut à un `user` côté e-commerce,
un `order` à un `purchase`, et un 'company' à un `business`."
```

## Concepts généraux

L’une des questions à se poser quand on implémente un CRM c’est au niveau de la gestion des erreurs.

> Lors de notre implémentation, nous avons opté pour un Event Driven Design (lien vers l’article de Marie), asynchrone, avec une politique de retry.

Reviewer 2: ```suggestion
Lors de notre implémentation, nous avons opté pour un Event Driven Design ([lien vers l’article de Marie](https://blog.eleven-labs.com/fr/event-driven-architecture-examples/)), asynchrone, avec une politique de retry.
```

> Lors de notre implémentation, nous avons opté pour un Event Driven Design (lien vers l’article de Marie), asynchrone, avec une politique de retry.

Reviewer: ```suggestion
Lors de notre implémentation, nous avons opté pour un **Event Driven Design** (lien vers l’article de Marie), asynchrone, avec une politique de retry.
```

> Lors de notre implémentation, nous avons opté pour un Event Driven Design (lien vers l’article de Marie), asynchrone, avec une politique de retry.

Reviewer: ```suggestion
Lors de notre implémentation, nous avons opté pour un Event Driven Design (lien vers l’article de Marie), asynchrone, avec une politique de Retry.
```

For coherence, as it's a capital R further in the article

> ### Asynchrone:

Reviewer 2: ```suggestion
### Asynchrone
```

L'Event Driven Design offre une grande résilience.
Lorsqu'un événement est publié, il est stocké dans une file d'attente ou un système de messagerie.
Si, pour une raison quelconque, la synchronisation échoue, les événements restent dans la file d'attente, prêts à être traités à nouveau.
Cela signifie que les erreurs temporaires ou les pannes de système n'impactent pas la synchronisation des données.
De plus, la possibilité de mettre en place des mécanismes de retry automatique garantit que les données seront finalement synchronisées avec succès,
même en cas de problèmes temporaires.

![Schema Detailed]({BASE_URL}/imgs/articles/2023-11-29-integration-crm/integration-crm-schema-detailled.png)

> Dans notre exemple de synchronisation avec Hubspot, nous avons un webhook qui est trigger lorsqu’un statut spécifique de Company est ajouté côté CRM.
> Le besoin sur notre application est de récupérer les informations de Company et de Contacts associés au trigger, et de les stocker.

Reviewer 2: à mon avis il faut inverser les phrases pour mieux comprendre le contexte.

```suggestion
Le besoin sur notre application est de récupérer les informations de Company et de Contacts associés au trigger, et de les stocker.
Dans notre exemple de synchronisation avec Hubspot, nous avons un webhook qui est trigger lorsqu’un statut spécifique de Company est ajouté côté CRM.
```

Une solution simple et synchrone serait de GET ces informations via l’API d’Hubspot.

> Imaginons qu’un problème survient au moment de la synchronisation des Contacts de la Company en question, quel est l’état de notre base côté applicatif ?

Reviewer 2: ```suggestion
Imaginons qu’un problème survient au moment de la synchronisation des Contacts de la Company en question, quel est l’état de notre base de données côté applicatif ?
```

Nous aurons une Company enregistrée avec aucun contact, et une erreur dans les logs.
Aussi il faudrait re-GET les informations de contacts liés à cette Company.

> Même problème pour la Company, où en cas d’erreur sur notre application, nous devons refaire une call pour récupérer ces mêmes informations.

Reviewer: We say `une call` or `un call` ?

Reviewer 2: un appel

:D

une fois par jour est moins réactive, peut générer des problèmes de performance et peut
entraîner des retards dans la mise à jour des données.
Le choix entre ces deux approches dépend des besoins spécifiques de votre entreprise
et de vos objectifs en matière de gestion des données.

### Initialisation des données

Lorsque vous envisagez d'intégrer une nouvelle plateforme e-commerce à votre CRM existant,
ou lorsque vous migrez vers un nouveau CRM, l'une des étapes critiques est l'initialisation
des données.
Cette phase consiste à transférer l'ensemble des données historiques,
telles que les informations sur les clients, les contacts, et bien plus encore,
du CRM vers votre plateforme (ou autre CRM).
C'est là que la synchronisation par lots (batching) joue un rôle crucial.

Voici certains avantages que nous avons notés lors d’une synchronisation par batching :

- **Contrôle et Précision** : Lors de la migration initiale, il est essentiel d'avoir un contrôle
précis sur les données qui sont transférées.
Le batching permet de planifier soigneusement la migration, de réaliser des tests,
d'identifier et de résoudre les problèmes potentiels avant le transfert complet des données.

>   Minimisation des Perturbations : Les migrations initiales de données peuvent être une tâche complexe, surtout lorsque de grandes quantités de données doivent être déplacées. Le batching vous permet de minimiser les perturbations dans les opérations quotidiennes, car vous pouvez planifier la migration en dehors des heures de pointe.

Reviewer 2: ```suggestion
- **Minimisation des Perturbations** : Les migrations initiales de données peuvent être des tâches complexes, surtout lorsque de grandes quantités de données doivent être déplacées. Le batching vous permet de minimiser les perturbations dans les opérations quotidiennes, car vous pouvez planifier la migration en dehors des heures de pointe.
```

---
contentType: article
lang: fr
date: 2023-11-29
slug: integration-crm
title: "Comment intégrer son CRM en asynchrone"
excerpt: "Découvrez un cas d'usage d'intégration d'un CRM en asynchrone, avec Hubspot et RabbitMq, au sein d'une application e-commerce"
categories:
    - php
keywords:
    - crm
    - hubspot
    - rabbitmq
authors:
    - youyou
    - fbelet
    - charles-eric
---

> Dans cet article, nous allons nous plonger dans l'intégration d'un CRM (Customer Relationship Management)

Reviewer 3: ajouter mention du Studio (ex : article composant react avec varnish ESI) pour dire que c'est une rex d'un vrai cas client studio
